pyroomacoustics/room.py

```diff
@@ -1586,9 +1587,38 @@ def __init__(self,
 
         Room.__init__(self, walls, fs, t0, max_order, sigma2_awgn, sources, mics)
 
+    def __str__(self):
+        return f'ShoeBox instance in {self.dim}D with {len(self.walls)} walls'
+
     def extrude(self, height):
         ''' Overload the extrude method from 3D rooms '''
 
         Room.extrude(self, np.array([0., 0., height]))
 
+class InfiniteRoom(Room):
```

Comment:
Such a room is also something I have been thinking about for some time :) I'm not completely sure about the InfiniteRoom name though. I'm not sure that is what a user would immediately think about. I'm more in favor of AnechoicRoom or FreeSpace. But there might be yet a better name. The advantage of AnechoicRoom would be to retain the idea of a limitited space (a room), but without reverb.

Reply:
I agree it might not be the best name! I see a few issues:
- InfiniteRoom: indeed, the user might not understand immediately what this means
- AnechoicRoom: My favorite, however I would not make it an actual room, as in all objects having to be inside it, because that would seem like an unnecessary nuisance to the user? That's where my "InfiniteRoom" idea came from: to have walls at infinite distance, such that there is no echo effect, and one does not have to worry about placing objects inside it.
- FreeSpace: It is the most fitting term imo, however "FreeSpace is not a type of Room" so in that sense, it does not comply with inheritance best practices. If we remove inheritance, I would suggest to use this name.

Reply:
> AnechoicRoom: My favorite choice, however I would not make it an actual room with walls, with all objects having to be inside it. That would seem like an unnecessary nuisance to the user?

I'm definitely with you on that one. Regardless of the name, we should not require dimensions. If there's a rewrite of the class, I don't think that should be a big problem. I don't know which name would be easiest to understand for users. Maybe we can brainstorm a few more names
* `UnboundedSpace`: makes it explicit there are no walls
* `Space`: simple
* `UnboundedRoom`: far fetched, variation on unbounded space
* `NoReverbRoom`: factual, less technical than anechoic

Reply:
I would vote for UnboundedRoom, NoReverbRoom or AnechoicRoom if we are making a subclass of Room. Another idea I had today, but maybe a bit far-fetched:
- `NoRoom`: would also make explicit that there are no walls, while using the "Room" concept anyways

The other option I see after working a bit more on the code is to have the current `Room` class inherit form a new `Space` baseclass (or a different name of course). `Space` has microphones, soundsources, and physical attributes like the speed of sound, etc. The `Room` subclass adds to this space class the concept of walls. This would require more rewriting though, and I let you decide if you think it is worth it! Personally, I think it would lead to the conceptually cleanest code.

Regarding my usecase, I would then use the `Space` class only